doc/memos/rdm-draft-buschsieweke-timer-api.md

```diff
@@ -0,0 +1,201 @@
+- RDM: 2
+- Title: RIOT's High Level Timer API
+- Author: TODO: Add authors prior merging
+- Status: Active
+- Type: Consensus
+- Created: October 2018
+
+## Abstract
+
+This document presents use cases, basic requirements, and goals for a high level
+timer API to be used in RIOT.
+
+## Status
+
+This document is a product of the community of RIOT maintainers, and aims to
+represent the consensus within this community. The content of this document is
+licensed with a Creative Commons CC-BY-SA license.
+
+## Terminology
+
+This document uses the [RFC2119](https://www.ietf.org/rfc/rfc2119.txt)
+terminology and the following acronyms:
+
+- RTC: Real Time Clock
+- RTT: Real Time Timer
+
+Throughout this document, the following terms should be understood as:
+
+- Real Time Clock: A low power hardware clock that represents time broken up
+  into year, month, day, hour, minute, and second. It has a fixed resolution of
+  1 second and provides all mandatory features of the [RTC API](http://api.riot-os.org/group__drivers__periph__rtc.html)
+- Real Time Timer: A low power hardware clock that is very similar to the RTC,
+  but operations on a time representation using a single integer. Unlike the
+  RTC, the resolution can differ from 1 second. RTTs provide all mandatory
+  features of the [RTT API](http://api.riot-os.org/group__drivers__periph__rtt.html)
+- Software Timer: A mechanism to run code at a given point in time in the
+  future. The number of concurrently usable software timers is only limited by
+  their memory requirements and the load a system can handle.
+- System Time: Common time reference of the local device, usually as a single
+  integer counting the clock ticks passed since some reference point
+
+# 1. Introduction
+
+## Motivation
+
+Many applications and use cases for using timer hardware for various things
+ranging from measuring time, delaying the execution of a thread, or running
+code at specific points in time exists. Convenience, portability and ease of
+use are therefore essential for a user friendly embedded OS.
+
+Actual timer hardware however is difficult to use. One main reason is that
+timer hardware can differ significantly starting with the clock frequency,
+the timer width (8 bit, 16 bit, 24 bit, and 32 bit are all common), and their
+time representation. Another reason is that hardware timers only allow setting
```

Comment:
multiple capture compare registers?

Reply:
maybe change to "allow setting a (very) small number of alarm callbacks"?

Reply:
Done

Reply:
I updated the formulation to the given suggestion.